README.md

```diff
@@ -1,38 +1,137 @@
-eventtarget.js
-========
+# eventtarget
 
-#### JavaScript events for custom objects ####
+JavaScript events for custom objects
 
-### Usage ###
+## Installation
+
+### Browser
 
 ```html
 <script src="EventTarget.js"></script>
-<script>
+```
 
-	// Applying EventTarget to custom object
+### Node
 
-	var Car = function () {
+```shell
+npm install
+```
 
-		EventTarget.call( this );
+```js
+var EventTarget = require('eventtarget');
+```
 
-		this.start = function () {
+## Usage
 
-			this.dispatchEvent( { type: 'started', foo: 'bar' } );
+```html
+<script>
 
-		};
+// Adding events to custom object
 
-	};
+var Car = function () {
+  this.start = function () {
+    this.dispatchEvent({type: 'start', message: 'vroom vroom!'});
+  };
+};
 
+Object.assign(Car.prototype, EventTarget.prototype);
 
-	// Using events
+// Using events
 
-	var car = new Car();
-	car.addEventListener( 'started', function ( event ) {
+var car = new Car();
 
-		alert( 'vroom vroom!' );
+car.addEventListener('start', function (event) {
+  alert(event.message);
+});
 
-	} );
-	car.start();
+car.start();
 
 </script>
 ```
+
+## Constructor
+
+If you inherit from `EventTarget`, you can invoke the (non-standard)
+constructor with options. See `setOptions`.
+
+## Standard methods
+
+The following behave as with the standard `EventTarget` methods
+(as possible). Options supported include the standard options, `capture`
+(including expressed as a boolean), as well as `once` (which auto-removes
+the listener after execution) and `passive` (which prevents the listener
+from using `preventDefault` even if the event passed to the target was
+`cancelable`).
+
+-   `addEventListener(type, listener, options)` - These standard methods cannot
+    prevent execution of the non-standard
+    `addDefaultEventListener`/`addLateEventListener`-added methods.
+
+-   `removeEventListener(type, listener, options)`
+
+-   `dispatchEvent(ev)` - Will check as appropriate the event's `type`,
+    `bubbles` or `cancelable` (which can all be set with invocation of the
+    constructor, `Event`) or the readonly `eventPhase` and `defaultPrevented`.
+    The properties `target`, `currentTarget`, `eventPhase`, and
+    `defaultPrevented` will be set as appropriate.
+
+-   `on* = function (ev) {};` (e.g., `onclick`) - Can use `return false` to
+    prevent the default (but this will not stop propagation or stop immediate
+    propagation as per regular JavaScript behavior (unlike jQuery which stops
+    propagation)).
+
+## Custom methods
```

Comment:
I will NEVER accept extra functionality on this module that's not defined on the W3C specification.